src/filesystem.cpp

```diff
@@ -82,6 +82,19 @@ std::string as_norm_dir( const std::string &path )
 {
     fs::path dir = fs::u8path( path ) / fs::path{};
     fs::path norm = dir.lexically_normal();
+#ifdef _WIN32
+    {
+        // Windows has strict rules for file naming
+        fs::path valid = norm.root_path();
+        fs::path rel = norm.relative_path();
+        for( auto &it : rel ) {
+            std::string item = it.generic_u8string();
+            item = ensure_valid_file_name( item );
+            valid /= utf8_to_wstr( item );
```

Comment:
Just for info, I tried but it didn't work. It appends only the first path and not the following. At the end of the loop, `valid` is like `C:\Users` instead of full path. It did work with wide-char strings.